### PR TITLE
GetAllProvisionedUsernames RPC

### PR DESCRIPTION
**Pull request by joshblum**
Fixes a dev related bug with missing users and exposes a RPC call to `GetAllProvisionedUsernames`

cc @chrisnojima if you'd like any tweaks to the call

### CODE REVIEW

File: go/service/config.go [hunk 1]
```diff
@@ -117,6 +117,21 @@ func (h ConfigHandler) GetExtendedStatus(ctx context.Context, sessionID int) (re
...
+func (h ConfigHandler) GetAllProvisionedUsernames(ctx context.Context, sessionID int) (res keybase1.AllProvisionedUsernames, err error) {
+	defaultUsername, all, err := libkb.GetAllProvisionedUsernames(libkb.NewMetaContext(ctx, h.G()))
+	if err != nil {
+		return res, err
+	}
+	provisionedUsernames := make([]string, len(all))
+	for i, u := range all {
+		provisionedUsernames[i] = u.String()
+	}
+	return keybase1.AllProvisionedUsernames{
```
[the remark this answers is not in the document]
Comment (joshblum) [reply go/service/config.go]: as far as i can tell `defaultUsername` is never in `provisionedUsernames` when coming from our side but the `defaultUsername` username will always be set to the last user to have logged in.

[the remark this answers is not in the document]
Comment (chrisnojima) [reply go/service/config.go]: yeah it shouldn't work like this i think
provisonedUsernames should be all the usernames period
defaultUsername should be one of those usernames, period

[the remark this answers is not in the document]
Comment (joshblum) [reply go/service/config.go]: kk sgtm

**Comment by joshblum** [comment]
```
🏎 ⤵️ [-calling] keybase.1.config.getAllProvisionedUsernames 
{defaultUsername: "test1", provisionedUsernames: Array(2)}
defaultUsername:"test1"
provisionedUsernames:(2) ["test", "test1"]
__proto__:Object
````

File: go/service/config.go [hunk 2]
```diff
@@ -117,6 +117,35 @@ func (h ConfigHandler) GetExtendedStatus(ctx context.Context, sessionID int) (re
...
+		return res, err
+	}
+
+	// If the default is missing, fill it in from the first provisioned.
+	if defaultUsername.IsNil() && len(all) > 0 {
+		defaultUsername = all[0]
+	}
+
+	// Callers expect ProvisionedUsernames to contain the DefaultUsername, so
+	// we ensure it is here as a final sanity check before returning.
```
Comment (maxtaco) [inline go/service/config.go]: i'm surpised `GetAllProvisionedUsernames` doesn't include the current one.  Why not?

Comment (joshblum) [reply go/service/config.go]: https://github.com/keybase/client/blob/master/go/libkb/config.go#L312

Comment (maxtaco) [reply go/service/config.go]: now i remember. thanks!

**Pull request merged**